src/test/java/javax/jmdns/impl/JmDNSTest.java

```diff
@@ -557,4 +558,45 @@ public void testListMyServiceWithoutLowerCase() throws IOException, InterruptedE
             if (registry != null) registry.close();
         }
     }
+
+    @Test
+    public void shouldNotNotifyServiceListenersForServiceResolveAfterServiceRemoved() throws UnknownHostException, IOException, InterruptedException {
+        String firstType = "_test._type1.local.";
+        String secondType = "_test._type2.local.";
+
+        JmmDNS jmmdns = JmmDNS.Factory.getInstance();
```

Comment:
We are here in the JmDNSTest class, not the JmmDNSTest class - I'd hence assume you should also create JmDNS instances here, right?

Reply:
Sorry, for two years, I have to recall a lot of things. There a lot of things I do not remember. I will try to provide answers soon.

Reply:
I just took a look and instances of the JmDNS are actually created few lines below in a for loop for each InetAddress.